similarity/calc_sim.py

```diff
@@ -0,0 +1,78 @@
+import nltk
+from nltk.corpus import stopwords
+import numpy as np
+import pandas as pd
+from sklearn.cluster import KMeans
+import string
+
+
+# Reads csv as input with Pandas DataFrame
+def read_data(filename):
+  data = pd.read_csv(filename, skipinitialspace=True)
+  return data
+
+# map each word to an index
+def extract_words(data):
+  bag_of_words = {}
+  nltk.download('stopwords')
+  stop_words = set(stopwords.words('english'))
+  index = 0
+  for about in data['Mission']:
+    # remove all punctuations
+    for char in about:
+      if char in string.punctuation:
+        about = about.replace(char, ' ')
+    # convert to all lowercase
+    about = about.lower()
+    # map each word to its index
+    for word in about.split():
+      if word not in bag_of_words.keys() and word not in stop_words:
+        bag_of_words[word] = index
+        index += 1
+  return bag_of_words
+
+# turns every mission statement into an array of word count
+def generate_features(df, word_dict):
+  num_orgs = df.shape[0]
+  num_words = len(word_dict)
+  feature_matrix = np.zeros((num_orgs, num_words))
+  missions = df['Mission']
+  j = 0
+  for m in missions:
+    # remove all punctuations
+    for char in m:
+      if char in string.punctuation:
+        m = m.replace(char, ' ')
+    # convert to all lowercase
+    m = m.lower()
+    for word in m.split():
+      if word in word_dict.keys():
+        index = word_dict[word]
+        # ith column of word dict is the word in the jth review
+        feature_matrix[j][index] = 1
+    j += 1
+  return feature_matrix
+
+def main():
```

Comment:
Would love to see some example clusters somewhere, maybe just in a PR comment?

Reply:
[['JDRF', 'Amnesty International', 'March of Dimes', 'American Diabetes Association', 'Leukemia & Lymphoma Society', 'National Multiple Sclerosis Society', 'Mayo Clinic'], ['Doctors without Borders (Médecins Sans Frontières)'], ['TED', 'The Humane Society', 'Smithsonian', 'Monterey Bay Aquarium', 'Wounded Warrior Project', 'Best Friends Animal Society', 'Kiva', 'Livestrong', 'Invisible Children', 'Public Broadcasting System (PBS)', 'USO', 'Human Rights Campaign', 'National Wildlife Federation', 'Oxfam', 'charity water', 'American Heart Association', 'Environmental Defense Fund', 'New York Public Library', 'The Nature Conservancy', 'CARE', 'National Parks Conservation Association', 'Heifer International', 'ASPCA', 'Defenders of Wildlife', 'Creative Commons', 'Habitat for Humanity International', 'In Touch Ministries', 'Cleveland Clinic', 'The U.S. Fund for UNICEF', 'Teach for America', 'Save the Children', 'San Diego Zoo', 'Ducks Unlimited', 'Make-A-Wish', 'American Red Cross', 'Feeding America', 'Susan G Komen for the Cure', 'Audubon', 'The Rotary Foundation', 'AARP', 'Boy Scouts of America', 'NPR']]